org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractEditorDoubleClickTextSelectionTest.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2018, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.ui.testing;
+
+import java.lang.reflect.InvocationTargetException;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jface.text.ITextSelection;
+import org.eclipse.jface.text.source.ISourceViewer;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.ui.texteditor.AbstractTextEditor;
+import org.eclipse.xtend2.lib.StringConcatenation;
+import org.eclipse.xtext.resource.FileExtensionProvider;
+import org.eclipse.xtext.ui.editor.XtextEditor;
+import org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil;
+
+import com.google.inject.Inject;
+
+/**
+ * @author miklossy - Initial contribution and API
+ * 
+ * @since 2.14
+ */
+public abstract class AbstractEditorDoubleClickTextSelectionTest extends AbstractEditorTest {
+
+	@Inject
+	private FileExtensionProvider fileExtensionProvider;
+
+	/**
+	 * @return the special symbol indicating the current cursor position.
+	 */
+	public String c() {
+		return "|";
+	}
+
+	/**
+	 * @param charSequence
+	 *            The editor's input text. The text must contain the {@link #c} special symbol indicating the current cursor position.
+	 * 
+	 * @param expected
+	 *            The text that is expected to be selected after double clicking in the Xtext editor on the current cursor position.
+	 */
+	public void assertSelectedTextAfterDoubleClicking(CharSequence charSequence, String expected) {
+		IFile file = createFile(getContent(charSequence));
+		try {
+			XtextEditor xtextEditor = openEditor(file);
+			xtextEditor = doubleClick(xtextEditor, getCursorPosition(charSequence));
+			assertSelectedText(xtextEditor, expected);
+		} catch (Exception e) {
+			fail("Error opening editor associated with the file " + file.getName());
```

Comment:
Added this line to ensure that the test fails if an exception is thrown, and a proper error message is provided to the user.